code/modules/reagents/reagent_dispenser.dm

```diff
@@ -1,72 +1,102 @@
-
-
 /obj/structure/reagent_dispensers
-	name = "Dispenser"
-	desc = "..."
+	name = "Strange dispenser"
+	desc = "What the fuck is this?"
 	icon = 'icons/obj/objects.dmi'
 	icon_state = "watertank"
 	density = 1
 	anchored = 0
+	var/accept_any_reagent = TRUE
 
 	var/amount_per_transfer_from_this = 10
-	var/possible_transfer_amounts = list(10,25,50,100)
+	var/possible_transfer_amounts = list(5,10,15,25,30,50,60,100,120,250,300)
 	var/capacity = 1000
-	attackby(obj/item/weapon/W as obj, mob/user as mob)
+	var/can_tamper = TRUE
+	var/is_leaking = FALSE
+
+/obj/structure/reagent_dispensers/New()
+	var/datum/reagents/R = new/datum/reagents(capacity)
+	reagents = R
+	R.my_atom = src
+	if (!possible_transfer_amounts)
+		src.verbs -= /obj/structure/reagent_dispensers/verb/set_APTFT
+	..()
+
+/obj/structure/reagent_dispensers/examine(mob/user)
+	if(!..(user, 2))
 		return
+	user << "<span class='notice'>It contains:</span>"
+	if(reagents && reagents.reagent_list.len)
+		for(var/datum/reagent/R in reagents.reagent_list)
+			user << "<span class='notice'>[R.volume] units of [R.name]</span>"

```

Comment:
How can the user determine what reagents are in it from a glance?

Reply:
Oh, good point. Before it said "Contains x units of fuel" so I did the OBVIOUS thing, not the smart thing, and made it list all the reagents.